pkg/datamover/repository_server_test.go

```diff
@@ -0,0 +1,214 @@
+	err = createTestKopiaRepository(rss.s3Location, rss.cli, rss.namespace.GetName(), rss.pod)
+	c.Assert(err, IsNil)
+
+	// Start Kopia Repository Server
+	err = startTestKopiaRepositoryServer(rss.cli, rss.namespace.GetName(), rss.pod)
+	c.Assert(err, IsNil)
+
+	// Wait for Kopia Repository Server To Get Ready
+	err = waitForServerReady(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod, rss.tlsSecret)
+	c.Assert(err, IsNil)
+
+	// Add Test Client User in Kopia Repository
+	err = addTestUserInKopiaRepositoryServer(rss.cli, rss.namespace.GetName(), rss.pod, rss.user)
+	c.Assert(err, IsNil)
+
+	// Refresh Kopia Repository Server
+	err = refreshTestKopiaRepositoryServer(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod, rss.tlsSecret)
+	c.Assert(err, IsNil)
+
+	// Wait for Kopia Repository Server To Get Ready
+	err = waitForServerReady(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod, rss.tlsSecret)
+	c.Assert(err, IsNil)
+
+	// Set Kopia Repo Server Template Param
+	contentCacheMB, metadataCacheMB := kopiacmd.GetGeneralCacheSizeSettings()
+	rss.pod, err = rss.cli.CoreV1().Pods(rss.namespace.GetName()).Get(rss.ctx, rss.pod.GetName(), metav1.GetOptions{})
+	c.Assert(err, IsNil)
+	rss.repoServer = &param.RepositoryServer{
+		Name:      testRepoServerName,
+		Namespace: rss.namespace.GetName(),
+		ServerInfo: crv1alpha1.ServerInfo{
+			PodName:     rss.pod.GetName(),
+			ServiceName: rss.service.GetName(),
+		},
+		Username: rss.user,
+		Credentials: param.RepositoryServerCredentials{
+			ServerTLS:        *rss.tlsSecret,
+			ServerUserAccess: *rss.userAccessSecret,
+		},
+		ContentCacheMB:  contentCacheMB,
+		MetadataCacheMB: metadataCacheMB,
+		Address:         fmt.Sprintf("https://%s:%d", rss.pod.Status.HostIP, rss.service.Spec.Ports[0].NodePort),
+	}
+}
+
+func (rss *RepositoryServerSuite) connectWithTestKopiaRepositoryServer() error {
+	return repository.ConnectToAPIServer(
+		rss.ctx,
+		string(rss.tlsSecret.Data[kopia.TLSCertificateKey]),
+		testKopiaRepoServerAdminPassword,
+		defaultKopiaRepositoryHost,
+		rss.repoServer.Address,
+		rss.repoServer.Username,
+		rss.repoServer.ContentCacheMB,
+		rss.repoServer.MetadataCacheMB,
+	)
+}
+
+func (rss *RepositoryServerSuite) TestLocationOperationsForRepositoryServerDataMover(c *C) {
+	// Setup Test Data
+	sourceDir := c.MkDir()
+	filePath := filepath.Join(sourceDir, "test.txt")
+
+	cmd := exec.Command("touch", filePath)
+	_, err := cmd.Output()
+	c.Assert(err, IsNil)
+
+	targetDir := c.MkDir()
+
+	// Connect With Test Kopia RepositoryServer
+	err = rss.connectWithTestKopiaRepositoryServer()
+	c.Assert(err, IsNil)
+
+	// Test Kopia Repository Server Location Push
+	snapInfo, err := kopiaLocationPush(rss.ctx, defaultKopiaRepositoryPath, "kandoOutput", sourceDir, testKopiaRepoServerAdminPassword)
+	c.Assert(err, IsNil)
+
+	// Test Kopia Repository Server Location Pull
+	err = kopiaLocationPull(rss.ctx, snapInfo.ID, defaultKopiaRepositoryPath, targetDir, testKopiaRepoServerAdminPassword)
```

Comment:
Prasad if we look ijnto `kopiaLocationPull` command, we could see if we're using `usePipeParam` `-` while running `kando location pull` than only we are getting the file, else we get the kopia entry for the same file. So while testing this function instead of file we're getting kopia entry for that file so we can't match the restored file contents

Reply:
Raised a Bug for the same https://github.com/kanisterio/kanister/issues/2230